### PR TITLE
feat(emails): Mailer accept multiple emails Part 1

### PR DESCRIPTION
**Pull request by vbudhram**
This PR is phase 1 of https://github.com/mozilla/fxa-auth-server/pull/1729#issuecomment-289026998 and connects to #1672. It normalizes all `mailer.*` function params to `(emails, account, opts)`. This allows us to have a common interface to send email to multiple addresses.

@mozilla/fxa-devs r?

### CODE REVIEW

File: test/local/lib/senders/index.js [hunk 1]
```diff
@@ -164,7 +160,7 @@ describe('lib/senders/index', () => {
           .then(e => {
             email = e
             email._ungatedMailer.postVerifyEmail = sinon.spy(() => P.resolve({}))
-            return email.sendPostVerifyEmail(EMAIL, {})
+            return email.sendPostVerifyEmail(EMAIL, acct, {})
```
Comment (philbooth) [inline test/local/lib/senders/index.js]: Shouldn't `EMAIL` be an array here too?

File: lib/senders/index.js [hunk 2]
```diff
@@ -96,16 +96,35 @@ module.exports = function (log, config, error, db, translator, sender) {
...
+    }
+
+    function getSecondaryEmails(emailData) {
+      return emailData.reduce(function(list, email) {
+        if (! email.isPrimary) {
+          list.push(email.email)
+        }
+        return list
+      }, [])
+    }
```
Comment (philbooth) [inline lib/senders/index.js]: I don't see anything in the tests that properly exercises these two functions. Should we tweak some of the existing tests so they pass in the different email types here and assert that whatever is supposed to happen happens?

File: lib/senders/index.js [hunk 3]
```diff
@@ -96,16 +96,35 @@ module.exports = function (log, config, error, db, translator, sender) {
...
+
     senders.email = {
-      sendVerifyCode: function (account, code, opts) {
-        return getSafeMailer(account.email)
+      sendVerifyCode: function (emailData, account, opts) {
```
Comment (philbooth) [inline lib/senders/index.js]: I'm deep into bike-shedding territory here but I'm not a *huge* fan of the name `emailData` for these arguments, it doesn't quite suggest this-is-an-array-of-email-addresses to me. If it was *my* bike shed, I'd paint it more like `emails` or `emailAddresses` or something. :smile:

**Comment by philbooth** [comment]
@vbudhram, looks good! Couple of questions about tests inline, other than that I'm r+. 👍

File: lib/senders/index.js (hunk 2 above)
Comment (vbudhram) [inline lib/senders/index.js]: Done!

**Comment by vbudhram** [comment]
@philbooth Thanks!

**Pull request merged**